pyopenms_viz/_dataframe.py

```
@@ -0,0 +1,141 @@
+            for name, group in self._grouped:
+                yield name, DataFrameWrapper(group)
+    
+    def agg(self, func: dict) -> DataFrameWrapper:
+        """Aggregate using the specified functions."""
+        if self._is_polars:
+            agg_exprs = []
+            for col, agg_func in func.items():
+                if isinstance(agg_func, str):
+                    agg_exprs.append(pl.col(col).agg(agg_func))
+                else:
+                    agg_exprs.append(pl.col(col).agg(lambda x: agg_func(x.to_numpy())))
```

Ruff (0.8.2)

Function definition does not bind loop variable `agg_func`

(B023)